### PR TITLE
Ensemble view

### PR DESCRIPTION
**Pull request by sharkcooper**

### CODE REVIEW

File: FIRO_TSEnsembles/src/main/java/hec/ensemble/Ensemble.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public int getTimeCount(){
       int size= getTimeCount();
       for (int i = 0; i <size ; i++) {
         rval[i] =t;
-        t.plus(_configuration.getDuration());
+        t = t.plus(_configuration.getDuration());
```
Comment (ktarbet) [inline FIRO_TSEnsembles/src/main/java/hec/ensemble/Ensemble.java:47]: Nice Catch!

File: FIRO_TSEnsembles/src/main/java/hec/metrics/MetricTypes.java [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package hec.metrics;
 
 public enum MetricTypes {
-    SINGLE_VALUE,
-    ARRAY_OF_SINGLE_VALUE,
-    TIMESERIES_OF_SINGLE_VALUES,
-    ARRAY_OF_TIMESERIES;
+    SINGLE_VALUE,//all values of an ensemble condensed into one.
+    ARRAY_OF_SINGLE_VALUE, //all timestep values of a trace condensed to one as an array of values for each trace
```
Comment (HenryGeorgist) [inline FIRO_TSEnsembles/src/main/java/hec/metrics/MetricTypes.java:5]: this is now out of date, based on a commit i made in metrics-tables-test

File: FIRO_TSEnsembles/src/test/java/hec/ensemble/EnsembleTimeSeriesTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            RecordIdentifier tsid = new RecordIdentifier("Kanektok.SCRN2", "flow");
+            //get an ensemble time series
+            EnsembleTimeSeries ets = db.getEnsembleTimeSeries(tsid);
+            //create a computable statistic
+            MultiComputable test = new MultiStatComputable(new Statistics[] {MIN, MEAN, MAX});
+            //compute the statistics for the entire ensemble time series
+            MetricCollectionTimeSeries output = ets.iterateAcrossTimestepsOfEnsemblesWithMultiComputable(test);
+            //verify at the data is properly computing for a set of known values
+            float[] value1 = output.iterator().next().getValues()[3];
+            assertEquals(-4000, value1[0]);//min
```
Comment (HenryGeorgist) [inline FIRO_TSEnsembles/src/test/java/hec/ensemble/EnsembleTimeSeriesTest.java:56]: this is out of date i believe

File: ensemble-view/src/main/java/hec/ensembleview/EnsembleViewer.java [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+    }
+
+    private void addMembersToChart(EnsembleChart chart, float[][] vals, ZonedDateTime[] dates) throws ParseException {
+        for (int i = 0; i < vals.length; i++) {
+            chart.addLine(new LineSpec(vals[i], dates, new BasicStroke(1.0f), null, "Member " + (i + 1)));
+        }
+    }
+
+    private void addStatisticsToChart(EnsembleChart chart, float[][] stats, Statistics[] selectedStats, ZonedDateTime[] dates) throws ParseException {
+        for (int i = 0; i < selectedStats.length; i++) {
```
Comment (HenryGeorgist) [inline ensemble-view/src/main/java/hec/ensembleview/EnsembleViewer.java:143]: good idea.

File: ensemble-view/src/main/java/hec/ensembleview/EnsembleViewer.java [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+            switch (selectedStats[i]){
+                case MIN:
+                    chart.addLine(new LineSpec(stats[i], dates, new BasicStroke(3.0f), Color.BLACK, "MIN"));
```
Comment (HenryGeorgist) [inline ensemble-view/src/main/java/hec/ensembleview/EnsembleViewer.java:146]: can we add a dash array to this constructor (and the max constructor)

**Review by HenryGeorgist — APPROVED** [review]

**Pull request merged**